contract_manager/src/token.ts

```diff
@@ -0,0 +1,71 @@
+import axios from 'axios';
+import { KeyValueConfig, Storable } from "./base";
+
+/**
+ * A token represents a cryptocurrency like ETH or BTC.
+ * The main use of this class is to calculate the dollar value of accrued fees.
+ */
+export class Token extends Storable {
+  static type = "token";
+
+  public constructor(
+    public id: string,
+    // The hexadecimal pyth id of the tokens X/USD price feed
+    // (get this from hermes or the Pyth docs page)
+    public pythId: string | undefined,
+    public decimals: number,
+  ) {
+    super();
+  }
+
+  getId(): string {
+    return this.id;
+  }
+
+  getType(): string {
+    return Token.type;
+  }
+
+  /**
+   * Get the dollar value of 1 token. Returns undefined for tokens that do
+   * not have a configured pricing method.
+   */
+  async getPrice(): Promise<number | undefined> {
+    if (this.pythId) {
```

Comment:
not for now: maybe later for simplicity we can use Pyth symbol instead of pythId and get the mapping from Hermes too.